src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,261 @@ public static class NUnitTestExtractorApp
-                data = declaringType + "." + name + "()";
+                // match the "Exclude" category from the function category attibute list
+                // if the function has matched exclude category, we will skip this function
+                if (categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.ExcludeInfo))))
+                {
+                    continue;
+                }
+
+                // match the "Include" category from the function category attribute list
+                // if the function has matched include category, we will count this function as a valid test
+                if (string.IsNullOrWhiteSpace(_options.IncludeInfo)
+                    || categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.IncludeInfo))))
+                {
+                    tests.Add(test.DeclaringType.GetTypeInfo().FullName + "." + test.Name);
+                }
             }
 
-            return data;
+            return tests;
         }
 
         /// <summary>
-        /// Parses string into proper Level
+        /// Create the output file if it doesn't exist, or append it if it exists
         /// </summary>
-        /// <param name="levelToParse">The string to parse</param>
-        /// <returns>The Level from parsing the string</returns>
-        public static Level ParseLevel(string levelToParse)
+        /// <param name="outputInfo">The raw information need to be re-construct to our expected format.</param>
+        private static void OutputInfo(HashSet<string> outputInfoList)
         {
-            Level level = Level.Null;
-            try
-            {
-                level = (Level)Enum.Parse(typeof(Level), levelToParse, ignoreCase: true);
-            }
-            catch(ArgumentException e)
-            {
-                Console.Error.WriteLine(e.GetType() + ": Please enter either Namespace, Class, or Function for level");
-                Environment.Exit(-1);
-            }
-            return level;
-        }
+            ThrowIf.ArgumentNull(outputInfoList, nameof(outputInfoList));
 
-        /// <summary>
-        /// Appends all of a TestCase's arguments to a string
-        /// </summary>
-        /// <param name="args">array of arguments to append to string</param>
-        /// <param name="data">the string to add to and also return</param>
-        /// <returns>all of the arguments in a string with proper formatting</returns>
-        public static string AppendArgumentsForTestCasesToString(object[] args, string data)
-        {
-            if(args != null)
+            foreach (var outputInfo in outputInfoList)
             {
-                List<string> testArgs = new List<string>();
-                 
-                foreach(object arg in args)
-                {
-                    string argString = arg as string;
+                string finalOutput = StringUtils.FormatInvariant("\"" + _assembly.Location + "\"" + " | " + outputInfo);
```

Comment:
[Should] This is just using string concatenation, so there's nothing to Format.
Use this instead:
`string finalOutput = StringUtils.FormatInvariant("\"{0}\" | {1}", _assembly.Location, outputInfo);`